### ISSUE TITLE
nanoapi_generated.h should be generated in the build folder

### ISSUE THREAD

**Issue by dsiganos** [nanocurrency/nano-node#3498]
After the upgrade of flatbuffers library my build broke and even deleting the entire build directory didn't fix the problem. I think the problem was that nanoapi_generated.h needed refreshing. The build started succeeding after I deleted nanoapi_generated.h and I rebuilt everything. That makes me wonder, shouldn't nanoapi_generated.h be residing in the build directory?

I am not sure what the problem was exactly, all I know is that my build started succeeding when I deleted it. And my question is, since it is a build artifact, shouldn't it be in the build directory?

**Comment by dsiganos** [comment]
Relevant discussion on slack:
https://nanocurrency.slack.com/archives/C0220JV075L/p1633703163306400

**Comment by dsiganos** [comment]
@cryptocode , do you have any input on this?

**Comment by cryptocode** [comment]
@dsiganos Yeah, emitting it to the build dir sounds better. I don't recall if there was a reason for doing it the current way or not.
